From 860267bd6288dfe8ec6c662dc75af955504257c8 Mon Sep 17 00:00:00 2001
From: Michael Niedermayer <michael@niedermayer.cc>
Date: Sun, 4 Oct 2026 23:44:39 +0200
Subject: [PATCH] avformat/mov: accept ipma associations with property_index 0

ISO/IEC 23008-12 defines a property_index of 0 as no property being
associated, but mov_read_iprp() rejected it, and with it the whole file.

Found during triage of the security report srZp1wXh4CXQ

Signed-off-by: Michael Niedermayer <michael@niedermayer.cc>
---
 libavformat/mov.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/libavformat/mov.c b/libavformat/mov.c
index fb1b91772b49..a7400aba0696 100644
--- a/libavformat/mov.c
+++ b/libavformat/mov.c
@@ -9985,7 +9985,9 @@ static int mov_read_iprp(MOVContext *c, AVIOContext *pb, MOVAtom atom)
                 index <<= 8;
                 index |= avio_r8(pb);
             }
-            if (index > nb_atoms || index <= 0) {
+            if (!index)
+                continue;
+            if (index > nb_atoms) {
                 ret = AVERROR_INVALIDDATA;
                 goto fail;
             }
